analyzers/tests/SonarAnalyzer.UnitTest/Helpers/FilesToAnalyzeProviderTest.cs

```diff
@@ -0,0 +1,112 @@
+* along with this program; if not, write to the Free Software Foundation,
+* Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+*/
+
+using System.Text.RegularExpressions;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.UnitTest.Helpers
+{
+    [TestClass]
+    public class FilesToAnalyzeProviderTest
+    {
+        private const string MixedSlashesWebConfigPath1 = @"C:\Projects/DummyProj/wEB.config";
+        private const string MixedSlashesWebConfigPath2 = @"C:\Projects/DummyProj/Views\Web.confiG";
+        private const string FilesToAnalyzePath = @"ResourceTests\FilesToAnalyze\FilesToAnalyze.txt";
+        private const string InvalidFilesToAnalyzePath = @"ResourceTests\FilesToAnalyze\InvalidValuesFileToAnalyze.txt";
+
+        [TestMethod]
+        public void FileNameWithMixedCapitalizationAndMixedSlashes_FindFilesWithFileName_ReturnsAllWebConfigFiles()
+        {
+            var sut = new FilesToAnalyzeProvider(FilesToAnalyzePath);
+
+            var results = sut.FindFiles("Web.config");
+            results.Should().HaveCount(2);
+            results.Should().Contain(MixedSlashesWebConfigPath1);
+            results.Should().Contain(MixedSlashesWebConfigPath2);
+        }
+
+        [TestMethod]
+        public void FileNameWithMixedCapitalizationAndMixedSlashes_FindFilesWithRegex_ReturnsAllWebConfigFiles()
+        {
+            var fileNamePattern = new Regex("web\\.config$", RegexOptions.IgnoreCase);
+
+            var sut = new FilesToAnalyzeProvider(FilesToAnalyzePath);
+
+            var results = sut.FindFiles(fileNamePattern);
+            results.Should().HaveCount(2);
+            results.Should().Contain(MixedSlashesWebConfigPath1);
+            results.Should().Contain(MixedSlashesWebConfigPath2);
+        }
+
+        [TestMethod]
+        public void FileWithInvalidvalues_FindFilesWithFileName_ReturnsValidValue()
+        {
+            var sut = new FilesToAnalyzeProvider(InvalidFilesToAnalyzePath);
+
+            var results = sut.FindFiles("Web.config");
+            results.Should().HaveCount(1);
+            results.Should().Contain(MixedSlashesWebConfigPath2);
+        }
+
+        [TestMethod]
+        public void FileWithInvalidvalues_FindFilesWithRegex_ReturnsValidValue()
+        {
+            var fileNamePattern = new Regex("web\\.config$", RegexOptions.IgnoreCase);
+
+            var sut = new FilesToAnalyzeProvider(InvalidFilesToAnalyzePath);
+
+            var results = sut.FindFiles(fileNamePattern);
+            results.Should().HaveCount(1);
+            results.Should().Contain(MixedSlashesWebConfigPath2);
+        }
+
+        [TestMethod]
+        public void NonExistentFile_FindFiles_ReturnsEmptyEnumerable()
+        {
+            var sut = new FilesToAnalyzeProvider(@"ResourceTests\FilesToAnalyze\NonExistingFile.txt");
+
+            var results = sut.FindFiles("Web.config");
```

Comment:
Same the other one for empty
```suggestion
            var results = sut.FindFiles(new Regex(".*"));
```